contracts/FlowIDTableStaking.cdc

```diff
@@ -1004,6 +1024,21 @@ pub contract FlowIDTableStaking {
         }
     }
 
+    /// Gets the total amount of tokens that have been staked for a node. 
+    /// The sum from the node operator and all its delegators
+    pub fun getNodeStakedBalanceWithDelegators(_ nodeID: String): UFix64 {
+        let nodeRecord = self.borrowNodeRecord(nodeID)
+
+        var sum: UFix64 = nodeRecord.tokensStaked.balance
+
+        for delegator in nodeRecord.delegators.keys {
+            let delRecord = nodeRecord.borrowDelegatorRecord(delegator)
+            sum = sum + delRecord.tokensStaked.balance
```

Comment:
It doesn't help here, but this is the kind of record keeping that in future we should update with each transaction from (in this example) a delegator.

Reply:
yes, I am learning a lot about how to write better cadence code from this testing